File: app/src/main/java/com/example/thermostat/network/Repo.java
package com.example.thermostat.network;


import com.example.thermostat.Constants;
import com.example.thermostat.model.Bungalow;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.PUT;

public interface Repo {

    // get single
    @Headers("Content-Type:application/json; charset=UTF-8")
    @GET(Constants.idValue)
    Call<Bungalow> getSingleBungalow();

    // get all
    @Headers("Content-Type:application/json; charset=UTF-8")
    @GET(Constants.baseUrl)
    Call<List<Bungalow>> getAllBungalows();

    //put single
    @PUT(Constants.idValue)
    Call<Bungalow> putBungallow(@Pa)

}

File: app/src/main/java/com/example/thermostat/network/Repo.kt
package com.example.thermostat.network

import com.example.thermostat.Bungalow
import retrofit2.Response

class Repo(
    private val service: BungalowService

) : BungalowService {

    override suspend fun getSingleBungalow(): Response<Bungalow> {
        service.getSingleBungalow()
    }

    override suspend fun getBungalows(): Response<List<Bungalow>> {
        service.getBungalows()
    }


}